frontend/src/components/shared/task-form.tsx

```diff
@@ -57,21 +64,37 @@ export const TaskForm = React.forwardRef<HTMLFormElement>((_, ref) => {
     return "What do you want to build?";
   }, [selectedRepository]);
 
-  const handleSubmit = (event: React.FormEvent<HTMLFormElement>) => {
+  const handleSubmit = async (event: React.FormEvent<HTMLFormElement>) => {
     event.preventDefault();
     const formData = new FormData(event.currentTarget);
 
     const q = formData.get("q")?.toString();
-    if (q) dispatch(setInitialQuery(q));
+    if (!q) return;
+
+    dispatch(setInitialQuery(q));
 
-    posthog.capture("initial_query_submitted", {
-      entry_point: "task_form",
-      query_character_length: q?.length,
-      has_repository: !!selectedRepository,
-      has_files: files.length > 0,
-    });
+    try {
+      setIsInitializing(true);
+      // Initialize the session before navigating
+      const { conversation_id: conversationId } = await OpenHands.initSession({
+        githubToken: gitHubToken || undefined,
+        selectedRepository: selectedRepository || undefined,
+        args: settings || undefined,
+      });
 
-    navigate("/app");
+      posthog.capture("initial_query_submitted", {
+        entry_point: "task_form",
+        query_character_length: q.length,
+        has_repository: !!selectedRepository,
+        has_files: files.length > 0,
+      });
+
+      navigate(`/conversation/${conversationId}`);
+    } catch (error) {
+      // TODO: Show error toast
+    } finally {
+      setIsInitializing(false);
+    }
```

Comment:
We could move this into a `useMutation` hook: https://tanstack.com/query/latest/docs/framework/react/guides/mutations